### ISSUE TITLE
IE10+: LocalStorage + token-authenticator causes infinite restore() calls

### ISSUE THREAD

**Issue by mize85** [simplabs/ember-simple-auth#469]
Hi marcoow,

First of all: Thanks for the great ember-simple-auth and all your work!

I ran in an wiered issiue on ie10 today. The restore() function from Authenticator was constantly called.

I was able to track it down to the util Function "objectsAreEqual" which always returns "false" if objects contain arrays (see: http://jsbin.com/qovofozije/1/edit?html,js,output). 

That fired always the "sessionDataUpdated" which causes the restore() to run again and again.
The Server returns some Permissions in Arrays at this point - so maybe this usecase is not toooo exotic.

**Comment by marcoow** [comment]
Yeah, should definitely be supported. I guess the library needs a more stable equality checker (pull request welcome of course).

**Comment by michaelBenin** [comment]
I ran into this issue last week on this project but it was not from IE10.

It only happened when two windows were open in chrome and the response from restore was different from the response from authenticate. 

I've solved this by using a lodash deep equal in restore as well as making sure the response from both authenticate and restore are exactly the same. 

This would only happen with 2 windows open. This week I'll check my app against ie10 to recreate. 

This could be a problem within your app (like with mine) that restore and authenticate return different responses.

**Comment by tehmaestro** [comment]
So what would be the fix for this?

**Comment by marcoow** [comment]
@tehmaestro: that isn't really reliable as well as it depends on the sorting of the keys (which I think might be non-deterministic?).

**Comment by tehmaestro** [comment]
Ok, I've included the underscore js library, and I'm doing this:
```
      b = b || {};
      a = a || {};
      return _.isEqual(a, b);
```

The first 2 lines were needed, because I'm gotten cases where b would be `undefined`, so the comparison between `undefined` and `{}` would have returned false ... which is not what we want.

**Comment by marcoow** [comment]
I don't think Ember Simple Auth should depend on underscore.js.

**Comment by tehmaestro** [comment]
I agree, it's just my quick way of getting around this, so I can do some testing in Internet Explorer.

**Comment by marcoow** [comment]
Sure, maybe underscore.js's `isEqual` method could be extracted to replace [Ember Simple Auth's `objectsAreEqual`](https://github.com/simplabs/ember-simple-auth/blob/master/packages/ember-simple-auth/lib/simple-auth/utils/objects-are-equal.js)?

**Comment by tehmaestro** [comment]
I tried it, but `isEqual` calls at least another 2 functions from `underscore`. So there would be multiple things to bring over ..

**Comment by marcoow** [comment]
should be fixed by 44d563592b0114a6d16dc22f1937def690d971b6 - will release that with 0.8.0.beta.2
